### PR TITLE
Improve ContentFlow logs/docs

### PR DESCRIPTION
**Pull request by tonyr59h**
This is probably the biggest diff to deal with. Luckily it's mostly non-functional changes.

### CODE REVIEW

File: components/ContentFlow.brs [hunk 1]
```diff
@@ -145,62 +152,186 @@ sub onTruexEvent(event as object)
...
+
+        ? "TRUE[X] >>> ContentFlow::onTruexEvent() - user requested video stream playback cancel..."
         m.top.event = { trigger: "cancelStream" }
     end if
 end sub
 
-sub onPayload()
-    decodedData = m.sdkLoadTask.payload
+'--------------------------------------------------------------------------------------------------------
+' Callback triggered when ImaSdbTask updates its 'truexAdData' field. The following fields are expected:
```
Comment (colejona) [inline components/ContentFlow.brs]: ImaSdbTask typo

File: components/ContentFlow.brs [hunk 2]
```diff
@@ -145,62 +152,186 @@ sub onTruexEvent(event as object)
...
+    decodedData = event.getData()
     if decodedData = invalid then return
+
+    '
+    ' [4]
+    '
+    ? "TRUE[X] >>> ContentFlow::onTruexAdDataReceived() - seeking video position past placeholder ad and pausing..."
+
+    ' pause the stream, which is currently playing a video ad
+    m.videoPlayer.control = "pause"
```
Comment (colejona) [inline components/ContentFlow.brs:200]: should make sure it makes sense to pause at this point -- we might have to pause earlier: if an ad break is supposed to occur at a certain point, we wouldn't want to play through that point in the stream before inserting the ad

File: components/ContentFlow.brs [hunk 3]
```diff
@@ -145,62 +152,186 @@ sub onTruexEvent(event as object)
...
+    ' seek past the True[X] placeholder video ad
+    m.videoPlayer.seek = m.videoPlayer.position + decodedData.currentAdBreak.duration
```
Comment (colejona) [inline components/ContentFlow.brs:202]: shouldn't this seek only happen when `adFreePod` is observed in the future?

Comment (tonyr59h) [reply components/ContentFlow.brs:202]: Nope! This `seek` happens as soon as the ad break starts to skip past the true[X] placeholder video. The other `seek` will skip the remaining video ads in the pod.

**Review by colejona — APPROVED** [review]

**Pull request merged**